src/protocol/context/malicious.rs

```diff
@@ -101,11 +102,17 @@ impl<'a, F: Field> Context<F> for MaliciousContext<'a, F> {
         }
     }
 
-    fn prss(&self) -> Arc<IndexedSharedRandomness> {
-        self.inner.prss.indexed(self.step())
+    fn with_prss<T>(&self, handler: impl FnOnce(&Arc<IndexedSharedRandomness>) -> T) -> T {
+        let _span =
+            telemetry::metrics::span!("mal_prss", step = self.step(), role = self.role()).entered();
+        let prss = self.inner.prss.indexed(self.step());
+        handler(&prss)
```

Comment:
span is dropped after this line which means every `generate_...` call is executed within this span's context. That allows passing down step and role labels for proper metric breakdown

Reply:
yea I did not like it either. I can make PRSS aware of role which it would use only to emit metrics. For some reason I thought it would be a more questionable implementation because when we run this on a separate machine, the role will always be the same.

Ideally these things should be available to consume from some global state. I can try to use https://docs.rs/tokio/latest/tokio/task/struct.LocalKey.html for that - if I fail I would switch to propagate role to PRSS